File: scala/src/main/scala/map.scala
import com.jambit.practicalresult.*

@main def map: Unit =
  val result: Either[String, Seq[Int]] = listService(4)

  println(result)

  val mapped: Either[String, String] = result.map(list => list.reverse)
                                             .map(list => list.mkString("-"))

  println(mapped)

File: typescript/src/map.ts
import {listService} from "./listService";
import {Result} from "for-comprehension-ts";

const result: Result<number[], string> = listService(4)

console.log(result)

const mapped: Result<string, string> = result.map(list => list.reverse())
                                             .map(list => list.join("-"))

console.log(mapped)
